lib/Image/Adapter/Imagick.php

```diff
@@ -58,27 +58,24 @@ public function load($imagePath, $options = [])
         }
 
         // support image URLs
-        if (preg_match('@^https?://|^s3://@', $imagePath)) {
-            $tmpFilename = 'imagick_auto_download_' . md5($imagePath) . '.' . File::getFileExtension($imagePath);
-            $tmpFilePath = PIMCORE_SYSTEM_TEMP_DIRECTORY . '/' . $tmpFilename;
-
-            $this->tmpFiles[] = $tmpFilePath;
-
-            if (preg_match('@^s3://@', $imagePath)) {
-                $imageContent = file_get_contents($imagePath);
+        if (!stream_is_local($imagePath)) {
+            if (isset($options['asset'])) {
+                // imagick is only able to deal with local files
+                // if your're using custom stream wrappers this wouldn't work, so we create a temp. local copy
+                $imagePath = $options['asset']->getTemporaryFile();
             } else {
-                $imageContent = \Pimcore\Tool::getHttpData($imagePath);
-            }
+                $tmpFilename = 'imagick_auto_download_'.md5($imagePath).'.'.File::getFileExtension($imagePath);
+                $tmpFilePath = PIMCORE_SYSTEM_TEMP_DIRECTORY.'/'.$tmpFilename;
 
-            File::put($tmpFilePath, $imageContent);
-            $imagePath = $tmpFilePath;
-        }
+                $this->tmpFiles[] = $tmpFilePath;
 
-        if (!stream_is_local($imagePath) && isset($options['asset'])) {
-            // imagick is only able to deal with local files
-            // if your're using custom stream wrappers this wouldn't work, so we create a temp. local copy
-            $imagePath = $options['asset']->getTemporaryFile();
-            $this->tmpFiles[] = $imagePath;
+                $src = fopen($imagePath, 'rb');
```

Comment:
Hmm, how can we care about those temporary files? I do not want to implement `__destruct()` in a trait. And otherwise if we implement the cleanup in a separate method, we always have to think of the cleanup. Perhaps it would be better to use a separate class like 
```php
class TemporaryFileHelper {
  private $tmpFiles = [];
  
  public function __destruct() {
    foreach ($this->tmpFiles as $tempFile) {
            if (file_exists($tempFile)) {
                @unlink($tempFile);
            }
        }
  }

  public function getLocalFile($path) {
    if (!stream_is_local($path)) {
            $tmpFilename = 'tmpfile_'.md5($path).'.'.File::getFileExtension($path);
            $tmpFilePath = PIMCORE_SYSTEM_TEMP_DIRECTORY.'/'.$tmpFilename;

            self::tmpFiles[] = $tmpFilePath;

            $src = fopen($path, 'rb');
            $dest = fopen($tmpFilePath, 'wb', false, File::getContext());
            stream_copy_to_stream($src, $dest);
            fclose($dest);

            return $tmpFilePath;
        }

       return $path;
  }
}